### PR TITLE
Add support for negative numbers

### PR DESCRIPTION
**Pull request by alecnunn**
Adds support for importing negative numbers.
image

### CODE REVIEW

File: magic_numbers.py [hunk 1]
```diff
@@ -81,13 +81,17 @@ def _parse_float_name(name: str) -> Union[float, None]:
 	return float(f"{int_part}.{decimal_str}")
 
 def __getattr__(name: str) -> Union[int, float]:
+	is_negative = False
+	if name.startswith('NEGATIVE_'):
```
Comment (DavidBuchanan314) [inline magic_numbers.py]: could you add a check for `MINUS_` too?

File: magic_numbers.py [hunk 2]
```diff
@@ -81,13 +81,17 @@ def _parse_float_name(name: str) -> Union[float, None]:
...
 		val = _parse_float_name(name)
 		if val is not None:
-			return val
+			return val if not is_negative else -val
 	# fall back to integer parsing
 	val = _text2int(name.lower().replace('_', ' '))
 	if val is None: # should raise an appropriate AttributeError
 		return object.__getattribute__(_orig_module, name)
-	return val
+	return val if not is_negative else -val
```
Comment (DavidBuchanan314) [inline magic_numbers.py]: please flip these ternary operators i.e. `-val if is_negative else val`

File: magic_numbers.py (hunk 1 above)
Comment (alecnunn) [reply magic_numbers.py]: Absolutely. Will do in the morning.

File: magic_numbers.py [hunk 3]
```diff
@@ -81,13 +81,20 @@ def _parse_float_name(name: str) -> Union[float, None]:
 	return float(f"{int_part}.{decimal_str}")
 
 def __getattr__(name: str) -> Union[int, float]:
+	is_negative = False
+	if name.startswith('NEGATIVE_') or name.startswith('MINUS_'):
+		is_negative = True
+		if name[0] == 'N':
+			name = name[9:] # remove "NEGATIVE_"
+		elif name[0] == 'M':
+			name = name[6:] # remove "MINUS_"
```
Comment (DavidBuchanan314) [inline magic_numbers.py]: one more nit - 9 and 6 here are, ironically, magic numbers. we could avoid this (and simplify in general) by doing `name = name.partition("_")[2]`

**Comment by DavidBuchanan314** [comment]
nice, ty!

**Pull request merged**